File: Tp1/Palindromo/Palindromo.java
import java.util.Scanner;
public class Palindromo {
	public static boolean isPalindromo(String frase){
		int tam = frase.length() - 1;
		boolean result = true;
		for(int i = 0; i < frase.length(); i++){
			if(frase.charAt(i) == frase.charAt(tam)){
				result = result && true;
			} else {
				result = false;
			}
			tam--;
		}
		return result;
	}

	public static boolean isFim(String frase){
		return (frase.length() == 3 && frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M');
	}

	public static void main(String args[]){
		Scanner entrada = new Scanner(System.in);

		String frase = entrada.nextLine();
		while(!isFim(frase)){
			if(isPalindromo(frase)){
				System.out.println("SIM");
			} else {
				System.out.println("NAO");
			}
			frase = entrada.nextLine();
		}
	}
}

File: Tp1/Palindromo/Palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool isPalindromo(char frase[100]){
	int x = strlen(frase) - 2;
	bool result = false;
	for(int i = 0; i < strlen(frase) - 1; i++){
		if(frase[i] == frase[x]){
			result = true;
		} else {
			result = false;
			break;
		}
		x--;
	}
	return result;
}

bool isFim(char frase[100]){
	return (frase[0] == 'F' && frase[1] == 'I' && frase[2] == 'M');
}

int main(void){
	char frase[1000];
	fgets(frase, 1000, stdin);
	fflush(stdin);
	while(!isFim(frase)){
		if(isPalindromo(frase)){
			printf("SIM\n");
		} else {
			printf("NAO\n");
		}
		fgets(frase, 1000, stdin);
	}
		
	return 0;	
}
